src/js/models/render-tree.js

```diff
@@ -12,6 +12,12 @@ export default class RenderTree {
   get rootNode() {
     return this._rootNode;
   }
+  /**
+   * @return {Boolean}
+   */
+  get isDirty() {
```

Comment:
This seems a bit funky, but it make sense with our current algo. In theory a branch of the tree could be dirty and the root node could not be dirty, and this logic would not catch that case.

Practically, today we always walk and dirty parent nodes anyway.

Basically I have a mild concern this would make a later refactoring of the render-tree complicated by implying dirty semantics based on the root node.